src/portfft/common/logging.hpp

```diff
@@ -28,13 +28,61 @@
 
 namespace portfft::detail {
 
+struct logging_config {
+  bool log_transfers = true;
+  bool log_dumps = true;
+  bool log_trace = true;
+  bool log_warnings = true;
+  logging_config() {
+    char* log_transfers_str = getenv("PORTFFT_LOG_TRANSFERS");
+    if (log_transfers_str != nullptr) {
+      log_transfers = static_cast<bool>(atoi(log_transfers_str));
+#ifndef PORTFFT_LOG_TRANSFERS
+      if (log_transfers) {
+        std::cerr << "Can not enable logging of transfers if it is disabled at compile time." << std::endl;
+      }
+#endif
+    }
+    char* log_dumps_str = getenv("PORTFFT_LOG_DUMPS");
+    if (log_dumps_str != nullptr) {
+      log_dumps = static_cast<bool>(atoi(log_dumps_str));
+#ifndef PORTFFT_LOG_DUMPS
+      if (log_dumps) {
+        std::cerr << "Can not enable logging of dumps if it is disabled at compile time." << std::endl;
+      }
+#endif
+    }
+    char* log_trace_str = getenv("PORTFFT_LOG_TRACE");
```

Comment:
Why do we need to check for the environment variable `PORTFFT_LOG_TRACE`, if its already mentioned at build ? It seems a bit redundant to set both. As this is only for tracing of the host side function calls. I would suggest removing Cmake option `PORTFFT_LOG_TRACES`, and let us rely on the environment variable only. It will be much more usable that way, as a special build for host trace will not be required

Reply:
Logging affects performance so we need a way do disable it at compile time.